src/lib/db-listeners/db-supports-listener.ts

```diff
@@ -0,0 +1,49 @@
+import { DB } from "../db";
+import { SupportItemType,
+         SupportAudienceType,
+         TeacherSupportModel,
+         TeacherSupportModelType } from "../../models/stores/supports";
+
+export class DBSupportsListener {
+  private db: DB;
+  private supportsRef: firebase.database.Reference | null = null;
+
+  constructor(db: DB) {
+    this.db = db;
+  }
+
+  public start() {
+    this.supportsRef = this.db.firebase.ref(
+      this.db.firebase.getSupportsPath(this.db.stores.user)
+    );
+    this.supportsRef.on("child_changed", this.handleSupportsUpdate);
+    this.supportsRef.on("child_added", this.handleSupportsUpdate);
+  }
+
+  public stop() {
+    if (this.supportsRef) {
+      this.supportsRef.off("child_changed", this.handleSupportsUpdate);
+      this.supportsRef.off("child_added", this.handleSupportsUpdate);
+    }
+  }
+
+  private handleSupportsUpdate = (snapshot: firebase.database.DataSnapshot) => {
```

Comment:
My ultimate plan is to have 3 listeners - one for classes, one for groups, and one for the specific student. Whenever anything under those keys changes, I'll clear and recreate the MST state that tracks it. It's a little wasteful, but I figure it's fine given the size/frequency of the updates.